urid/src/urid.rs

```diff
@@ -0,0 +1,54 @@
+use crate::feature::*;
+use core::UriBound;
+use std::cmp::PartialEq;
+use std::ffi::CStr;
+use std::ops::Deref;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct URID(u32);
```

Comment:
You can use `std::num::NonZeroU32` here instead of `u32`, it abstracts away a little bit of the work, and more importantly this way you'll get the memory layout optimization! :slightly_smiling_face:  (i.e. `Option<URID>` will have the same size as `u32`)

Reply:
Apart from the layout transparency (which I added), `NonZeroU32` doesn't do much, as you can see from the [sources](https://doc.rust-lang.org/src/core/num/mod.rs.html#53). However, it does not implement `Deref`, which is a shame since it makes so many things easier. Due to this reason, and that passing all method calls to `NonZeroU32` takes almost the same amount of code, I decided to implement it on my own.

Reply:
Making it transparent is not enough sadly :/ ([see this playground link for example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5d66a59efc8051d6e882e9dfde09a63b))

The bit that makes the magic in `NonZeroU32` is the `#[rustc_layout_scalar_valid_range_start(1)]` attribute, which is obviously not usable outside of it.

I can see why it does not implement `Deref`, since I believe this trait is made only for smart-pointers and the like.
Not implementing this trait sure makes comparisons and calculations with normal `u32`s more obnoxious, but considering most of the time URIDs are going to be just taken from `Map` and used as-is somewhere else, I don't think there are many use-cases for these manipulations, outside of our own tests for the `lv2-urid` crate, are there?

Reply:
Okay, I can see that. Looking at [the book](http://lv2plug.in/book/), the most common operation on URIDs are comparisons. I initially thought that implementing `Deref` for `URID` would make it comparable to `u32`s, but somehow it does not and I had to manually implement `PartialEq` for them. I will do the change now.